ui/src/main/java/edu/wpi/grip/ui/pipeline/StepController.java

```diff
@@ -82,9 +100,21 @@ private void initialize() {
         new Image(InputStream.class.cast(icon))));
     buttons.getChildren().add(0, exceptionWitnessResponderButtonFactory.create(step, "Step Error"));
 
+    if (step.getInputSockets().stream()
+        .allMatch(inputSocket -> inputSocket.getSocketHint().getView()
+            .equals(SocketHint.View.NONE))) {
+      expand.setManaged(false);
+    } else {
+      expandIcon.setImage(new Image("/edu/wpi/grip/ui/icons/up.png"));
+    }
+
     // Add a SocketControlView for each input socket and output socket
     for (InputSocket<?> inputSocket : step.getInputSockets()) {
-      inputSocketMapManager.add(inputSocketControllerFactory.create(inputSocket));
+      InputSocketController tempSocket = inputSocketControllerFactory.create(inputSocket);
+      inputSocketMapManager.add(tempSocket);
+      if (!inputSocket.getSocketHint().getView().equals(SocketHint.View.NONE)) {
```

Comment:
This is correct.  We only want to collapse User intractable sockets (not: TEXT, SLIDER, RANGE, SELECT, CHECKBOX)